### ISSUE TITLE
Warn .rubocop.yml file does not exist. Ignoring...

### ISSUE THREAD

**Issue by micahlisonbee** [misogi/vscode-ruby-rubocop#62]
Just upgraded to VS Code 1.19.1 today and the .rubocop.yml can't be found. It's in my project root. I'm on macOS 10.12.6.

This warning appears on every file load:
`.rubocop.yml file does not exist. Ignoring...`

My path setting is:
`"ruby.rubocop.configFilePath": ".rubocop.yml",`

I've also tried the full local path, no luck:
`"ruby.rubocop.configFilePath": "/Users/micliz/code/ravn/clients/beynd/beynd-api/.rubocop.yml"`

**Comment by micahlisonbee** [comment]
Update: Setting the full path now works, but the relative path `"ruby.rubocop.configFilePath": ".rubocop.yml",` still does not work. Still on `1.19.1`.

**Comment by psukys** [comment]
Can confirm that it as well does not work, if vscode variable is set up: `{workspaceRoot}`

**Comment by cannyedge34** [comment]
Same problem!

**Comment by thesimpledev** [comment]
Same problem..

**Comment by Kotwic4** [comment]
Same problem

**Comment by hafrei** [comment]
Same problem, but on Windows 10

**Comment by zismailov** [comment]
same problem

**Comment by MartinSGill** [comment]
Same problem.

> Can confirm that it as well does not work with vscode variable is set up: `{workspaceRoot}`

Variables are not supported in Settings Files.
https://code.visualstudio.com/docs/editor/variables-reference#_is-variable-substitution-supported-in-user-and-workspace-settings

**Comment by jlurena** [comment]
Is this issue being looked at?

**Comment by Rodrigue-K** [comment]
Can we confirm that this has not been resolved?

**Comment by estani** [comment]
Added a fix, waiting for acceptance - https://github.com/misogi/vscode-ruby-rubocop/pull/100

If you can't wait, here's the patch
[rubocop.js.patch.txt](https://github.com/misogi/vscode-ruby-rubocop/files/3256895/rubocop.js.patch.txt)

just apply it to the rubocop extension (e.g. ~/.vscode/extensions/misogi.ruby-rubocop-0.8.0/out/src/rubocop.js)

If the file was there you could do something like:
`patch ~/.vscode/extensions/misogi.ruby-rubocop-0.8.0/out/src/rubocop.js rubocop.js.patch.txt`

**Comment by swiknaba** [comment]
@estani this fix now shows me a warning, that the rubcoop.yml from the relative path to my project is not found, and VSCode will use `.rubcop.yml` --also I am not sure, which file exactly is meant.

```
Found multiple files (.rubocop.yml, ~/Sites/foo/bar/.rubocop.yml) will use .rubocop.yml)
```
-> I am currently working on project `/foo/bar`; having different rubocop configs for different projects. Passing in the ABS path works, however, I do not want to change the abs path in the settings whenever I switch to another repo. (besides, I still get a different behavior even when setting the abs path -.-).

When running `bundle exec rubocop`, no offenses are shown, however, inside VSCode, I get a lot of linting errors.
My rubocpp.yml file inherits the settings from another rubocop-gem. These are my VSCode settings now:

```json
{
    "ruby.rubocop.executePath": "~/.rbenv/shims/",
    "ruby.rubocop.useBundler": true,
}
```

local rubocop config:

```yml
# .rubocop.yml

inherit_gem:
  rubocop-rails_config:
    - config/rails.yml
```

-> using this gem: https://github.com/toshimaru/rubocop-rails_config/#usage

**Comment by TSMMark** [comment]
The `Found multiple files` warning popup comes up all the time and I can't get it to stop unless I remove `ruby.rubocop.configFilePath` from workspace settings, which I can't do.

It's literally talking about the same file. Just one is relative and not expanded for some reason:

<img width="482" alt="Screen Shot 2022-12-08 at 11 08 35 AM" src="https://user-images.githubusercontent.com/4197823/206500202-54301ddd-9626-4d19-bf82-30c8b3ca1f26.png">

Couldn't this be solved by expanding the relative path to an absolute path, and dedupe before comparing and deciding if "Found multiple files"?
